pkg/reporting/pullRequestReport.go

```diff
@@ -0,0 +1,292 @@
+				}
+				countPolicyViolationComponent[policyViolationName][componentName]++
+				scanReport.VulnsTableValues[policyViolationName] = append(scanReport.VulnsTableValues[policyViolationName],
+					policyViolationVulns{
+						VulnerabilityID:    policyVulnerability.Name,
+						VulnerabilityHref:  policyVulnerability.Meta.Href,
+						VulnerabilityScore: getScore(policyVulnerability.ErrorMessage, "score") + " " + getScore(policyVulnerability.ErrorMessage, "severity"),
+						ComponentName:      componentName,
+					})
+			}
+		}
+
+	}
+	// sorting allPolicyViolations
+	sort.Slice(allPolicyViolations, func(i, j int) bool {
+		return allPolicyViolations[i] < allPolicyViolations[j]
+	})
+	// sorting componentNames
+	sort.Slice(componentNames, func(i, j int) bool {
+		return componentNames[i] < componentNames[j]
+	})
+
+	if allPolicyViolations != nil && componentNames != nil {
+		scanReport.Success = false
+		// MainTableHeaders
+		scanReport.MainTableHeaders = append(scanReport.MainTableHeaders, "Component name")
+		for _, policyViolationName := range allPolicyViolations {
+			scanReport.MainTableHeaders = append(scanReport.MainTableHeaders, policyViolationName)
+		}
+		// MainTableValues
+		for i := range componentNames {
+			scanReport.MainTableValues = append(scanReport.MainTableValues, []string{})
+			scanReport.MainTableValues[i] = append(scanReport.MainTableValues[i], componentNames[i])
+			for j := 1; j < len(scanReport.MainTableHeaders); j++ {
+				policyV := scanReport.MainTableHeaders[j]
+				comp := componentNames[i]
+				count := strconv.Itoa(countPolicyViolationComponent[policyV][comp])
+				scanReport.MainTableValues[i] = append(scanReport.MainTableValues[i], count)
+			}
+		}
+		// sorting VulnsTableValues
+		for _, item := range scanReport.VulnsTableValues {
+			sort.Slice(item, func(i, j int) bool {
+				if strings.Contains(item[i].VulnerabilityScore, "10.0") {
+					return true
+				} else if strings.Contains(item[j].VulnerabilityScore, "10.0") {
+					return false
+				}
+				if item[i].VulnerabilityScore > item[j].VulnerabilityScore {
+					return true
+				} else if item[i].VulnerabilityScore < item[j].VulnerabilityScore {
+					return false
+				}
+				return item[i].ComponentName < item[j].ComponentName
+			})
+		}
+		// sorting NotVulnsTableHeaders
+		for _, item := range scanReport.NotVulnsTableValues {
+			sort.Slice(item, func(i, j int) bool {
+				return item[i].ComponentName <= item[j].ComponentName
+			})
+		}
+
+	}
+
+	tmpl, err := template.New("report").Parse(rapidReportMdTemplate)
+	if err != nil {
+		return "", errors.New("failed to create Markdown report template err:" + err.Error())
+	}
+	buf := new(bytes.Buffer)
+	err = tmpl.Execute(buf, scanReport)
+	if err != nil {
+		return "", errors.New("failed to create Markdown report template err:" + err.Error())
+	}
+	// removing report files
+	err = os.RemoveAll(removeDir)
+	if err != nil {
+		log.Println("couldn't remove report file")
```

Comment:
It seems to me that is would be better to use package `github.com/SAP/jenkins-library/pkg/log` to write logs because it is used across project and it makes possible to specify log level.
```suggestion
		log.Entry().Warn("couldn't remove report file")
```